server/src/test/java/org/elasticsearch/action/support/replication/TransportReplicationActionTests.java

```diff
@@ -597,8 +600,9 @@ public void testPrimaryReference() throws Exception {
             }
         };
         TestAction.PrimaryShardReference primary = action.new PrimaryShardReference(shard, releasable);
-        final Request request = new Request();
-        Request replicaRequest = (Request) primary.perform(request).replicaRequest;
+        // TODO check if the following two lines are really unused or might have side effects
+        // final Request request = new Request();
+        // Request replicaRequest = (Request) primary.perform(request).replicaRequest;
```

Comment:
@ywelsch replicaRequest was used to test the primaryTerm() with "assertThat(replicaRequest.primaryTerm(), equalTo(primaryTerm));". Since that got removed, this part of the test can also go I guess. Does the test still perform its function after this? After all the primaryTerm gets mocked out somewhere above, but never asserted anywhere later? Could you take a look and comment?

Reply:
you can delete this test. It doesn't do anything meaningful.